### PR TITLE
[MM-20158] Show JQL representation of subscription in modal

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR displays the JQL version of the subscription at the bottom of the modal.

Note that this should always produce valid JQL.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-20158

#### Screenshots

image

### CODE REVIEW

File: webapp/src/utils/jira_issue_metadata.tsx [hunk 1]
```diff
@@ -248,3 +248,84 @@ export function isEpicIssueType(issueType: IssueType): boolean {
...
+
+function quoteGuard(s: string) {
+    if (s && s.includes(' ')) {
+        return `"${s}"`;
+    }
+
+    return s;
+}
+
+export function generateJQLStringFromSubscriptionFilters(issueMetadata: IssueMetadata, fields: FilterField[], filters: ChannelSubscriptionFilters) {
```
Comment (cpoile) [inline webapp/src/utils/jira_issue_metadata.tsx]: One more step towards reimplementing Jira, right? :)

Comment (crspeller) [reply webapp/src/utils/jira_issue_metadata.tsx]: I can't wait to have to re-implement this server side for some reason 😆

**Comment by aaronrothschild** [comment]
@mickmister Looks good, but could we encapsulate it in a box or something (like a code block).  It looks a bit lost amongst the UI IMO.

**Comment by asaadmahmood** [comment]
@aaronrothschild I've updated the styling for the JQL stuff.
localhost_8065_mattermost_channels_off-topic

**Comment by DHaussermann** [comment]
@mickmister I can build and test this now but, when I try the full make, I see the following. Are you able to repro this? There is nothing new available when I try a Git Pull

```
/Users/dylanhaussermann/go/src/github.com/mattermost/mattermost-plugin-jira/webapp/src/utils/styles.js
  15:6  error  Missing semicolon  semi
  16:2  error  Missing semicolon  semi

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! webapp@0.0.1 lint: `eslint --ignore-pattern node_modules --ignore-pattern dist --ignore-pattern tests --ext .js --ext .jsx --ext .ts --ext .tsx . --quiet`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the webapp@0.0.1 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/dylanhaussermann/.npm/_logs/2019-11-19T16_13_15_760Z-debug.log
make: *** [check-style] Error 1
```

**Comment by DHaussermann** [comment]
This is working as expected. From tha Mattermost side, I don't see any issues.


But, users will not always be able to take this output of the filters and paste it in Jira to run a search. I'm not sure if that's part of the goal here. There are at least a couple cases where that will not work...

1. If the selected query has labels or field names that match reserved words such as label of `public` 
You will get `Error in the JQL Query: 'Public' is a reserved JQL word. You must surround it in quotation marks to use it in a query. `

2. Queries that contain security level do not work in the UI for me (cloud and server) 
You will get `Field 'Security Level' does not exist or you do not have permission to view it. `

@aaronrothschild is the goal that you can always paste the JQL into the Jira UI?
@mickmister  I'm concerned that the case I found on issue 1. would also affect event delivery. Thjoughts?

**Comment by mickmister** [comment]
Instead of shooting for valid JQL, I suggest we have a disclaimer saying that it is an estimation of the resulting query, and that we don't actually use the JQP output. @aaronrothschild What do you think?

> You must surround it in quotation marks to use it in a query

We can do this, but the query looks much less readable with quotes everywhere. And I would rather keep it how it is rather than making a whitelist for things like `public`.

> Field 'Security Level' does not exist or you do not have permission to view it.

It may be called something different in JQL land. Are you able to use the security level field somehow in Jira's JQL search UI? @DHaussermann

**Comment by mickmister** [comment]
@DHaussermann lint issues are fixed now

**Comment by mickmister** [comment]
Added "Summary" label as Asaad suggested offline. Also added the "Filters" label. No functional changes.

image

**Comment by aaronrothschild** [comment]
> Instead of shooting for valid JQL, I suggest we have a disclaimer saying that it is an estimation of the resulting query, and that we don't actually use the JQP output. @aaronrothschild What do you think?
> 
> > You must surround it in quotation marks to use it in a query
> 
> We can do this, but the query looks much less readable with quotes everywhere. And I would rather keep it how it is rather than making a whitelist for things like `public`.
> 
> > Field 'Security Level' does not exist or you do not have permission to view it.
> 
> It may be called something different in JQL land. Are you able to use the security level field somehow in Jira's JQL search UI? @DHaussermann

Sorry, back from vacation now.  

I agree with you @mickmister that it shouldn't be "executable" as is, but should probably have a disclaimer saying "Approximate JQL Equivalent:".  They can paste it in and fix the errors if needed, but I think it provides some extra clarity around what the UI is presenting to the user, not meant to replace the JQL builder (which will always be better than what we can do).
cc @DHaussermann

**Comment by DHaussermann** [comment]
@mickmister or @aaronrothschild 
Sorry, it's not clear to me if we need a further change here. Is this `Summary` label instead of a more detailed disclaimer?

**Comment by mickmister** [comment]
@DHaussermann How about instead of "Summary", we put "Approximate JQL Output"? Or a caption can be placed below "Summary"

**Comment by DHaussermann** [comment]
@mickmister Yes I prefer your suggestion of `Approximate JQL Output`. That seem clear enough to me.

**Comment by mickmister** [comment]
@DHaussermann Header has been updated to show "Approximate JQL Output"

@aaronrothschild Do you think this is something we should explain in the documentation?

**Comment by aaronrothschild** [comment]
> @aaronrothschild Do you think this is something we should explain in the documentation?

@DHaussermann Yes, I've labeled this for needing documentation so we don't forget.  I'm starting docs tomorrow.

**Pull request merged**